### ISSUE TITLE
Uncaught TypeError: Cannot read property 'slice' of undefined

### ISSUE THREAD

**Issue by zzjin** [atom/find-and-replace#841]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.14.0-beta0 x64
**Electron**: 1.3.13
**OS**: Mac OS X 10.12.3
**Thrown From**: [find-and-replace](https://github.com/atom/find-and-replace) package 0.205.1

### Stack Trace

Uncaught TypeError: Cannot read property 'slice' of undefined

```
At /Applications/Atom Beta.app/Contents/Resources/app.asar/node_modules/find-and-replace/lib/project/match-view.js:31

TypeError: Cannot read property 'slice' of undefined
    at Function.module.exports.MatchView.content (/app.asar/node_modules/find-and-replace/lib/project/match-view.js:31:54)
    at /app.asar/node_modules/space-pen/lib/space-pen.js:156:31)
    at Function.View.buildHtml (/app.asar/node_modules/space-pen/lib/space-pen.js:123:10)
    at /app.asar/node_modules/space-pen/lib/space-pen.js:155:33)
    at /app.asar/node_modules/find-and-replace/lib/project/match-view.js:22:46)
    at ResultView.module.exports.ResultView.renderResult (/app.asar/node_modules/find-and-replace/lib/project/result-view.js:93:31)
    at call (/packages/file-icons/lib/utils/general.js:145:38)
    at /packages/file-icons/lib/consumers/find-and-replace.js:18:19)
    at /packages/file-icons/lib/utils/general.js:147:19)
    at ResultView.module.exports.ResultView.initialize (/app.asar/node_modules/find-and-replace/lib/project/result-view.js:74:19)
    at /app.asar/node_modules/space-pen/lib/space-pen.js:184:25)
    at /app.asar/node_modules/find-and-replace/lib/project/result-view.js:20:47)
    at ResultsView.module.exports.ResultsView.renderResults (/app.asar/node_modules/find-and-replace/lib/project/results-view.js:205:22)
    at ResultsView.module.exports.ResultsView.attached (/app.asar/node_modules/find-and-replace/lib/project/results-view.js:146:19)
    at /app.asar/node_modules/space-pen/lib/space-pen.js:164:63)
    at /app.asar/node_modules/space-pen/lib/space-pen.js:51:55)
    at /app.asar/src/pane-element.js:138:24)
    at Function.module.exports.Emitter.simpleDispatch (/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/app.asar/node_modules/event-kit/lib/emitter.js:129:28)
    at Pane.module.exports.Pane.setActiveItem (/app.asar/src/pane.js:300:22)
    at Pane.module.exports.Pane.activateItem (/app.asar/src/pane.js:452:21)
    at /app.asar/src/workspace.js:545:18
```

### Commands

```
     -0:26.8.0 fuzzy-finder:toggle-file-finder (atom-pane.pane.active)
     -0:23.9.0 core:confirm (input.hidden-input)
     -0:21.4.0 fuzzy-finder:toggle-file-finder (input.hidden-input)
     -0:18.2.0 core:confirm (input.hidden-input)
     -0:15.7.0 tree-view:reveal-active-file (div.line)
     -0:11.2.0 tree-view:remove (span.name.icon.php-icon)
     -0:04.6.0 core:copy (input.hidden-input)
     -0:04.3.0 project-find:show (input.hidden-input)
     -0:04.1.0 core:paste (input.hidden-input)
     -0:04.0 core:confirm (input.hidden-input)
```

### Non-Core Packages

```
atom-alignment 0.12.1 
atom-ctags 5.0.0 
auto-indent 0.5.0 
autocomplete-php 0.3.7 
file-icons 2.0.8 
filesize 2.0.2 
git-control 0.8.2 
git-plus 7.0.7 
gruvbox 1.0.5 
linter 1.11.21 
linter-jshint 3.0.2 
linter-php 1.3.1 
nucleus-dark-ui 0.10.22 
pretty-json 1.6.1 
project-plus 0.9.0 
script 3.14.0 
```

**Comment by Bjoernsen** [comment]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.14.0-beta0 x64
**Electron**: 1.3.13
**OS**: Ubuntu 14.04.2
**Thrown From**: [find-and-replace](https://github.com/atom/find-and-replace) package 0.205.1


### Stack Trace

Uncaught TypeError: Cannot read property 'slice' of undefined

```
At /usr/share/atom-beta/resources/app.asar/node_modules/find-and-replace/lib/project/match-view.js:31

TypeError: Cannot read property 'slice' of undefined
    at Function.module.exports.MatchView.content (/app.asar/node_modules/find-and-replace/lib/project/match-view.js:31:54)
    at /app.asar/node_modules/space-pen/lib/space-pen.js:156:31)
    at Function.View.buildHtml (/app.asar/node_modules/space-pen/lib/space-pen.js:123:10)
    at /app.asar/node_modules/space-pen/lib/space-pen.js:155:33)
    at /app.asar/node_modules/find-and-replace/lib/project/match-view.js:22:46)
    at ResultView.module.exports.ResultView.renderResult (/app.asar/node_modules/find-and-replace/lib/project/result-view.js:93:31)
    at call (/packages/file-icons/lib/utils/general.js:145:38)
    at /packages/file-icons/lib/consumers/find-and-replace.js:18:19)
    at /packages/file-icons/lib/utils/general.js:147:19)
    at ResultView.module.exports.ResultView.initialize (/app.asar/node_modules/find-and-replace/lib/project/result-view.js:74:19)
    at /app.asar/node_modules/space-pen/lib/space-pen.js:184:25)
    at /app.asar/node_modules/find-and-replace/lib/project/result-view.js:20:47)
    at ResultsView.module.exports.ResultsView.renderResults (/app.asar/node_modules/find-and-replace/lib/project/results-view.js:205:22)
    at ResultsView.module.exports.ResultsView.attached (/app.asar/node_modules/find-and-replace/lib/project/results-view.js:146:19)
    at /app.asar/node_modules/space-pen/lib/space-pen.js:164:63)
    at /app.asar/node_modules/space-pen/lib/space-pen.js:51:55)
    at /app.asar/src/pane-container-element.js:42:14)
    at Function.module.exports.Emitter.simpleDispatch (/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/app.asar/node_modules/event-kit/lib/emitter.js:129:28)
    at PaneContainer.module.exports.PaneContainer.setRoot (/app.asar/src/pane-container.js:163:20)
    at PaneContainer.module.exports.PaneContainer.replaceChild (/app.asar/src/pane-container.js:173:19)
    at Pane.module.exports.Pane.split (/app.asar/src/pane.js:919:21)
    at Pane.module.exports.Pane.splitDown (/app.asar/src/pane.js:907:19)
    at showFile (/packages/git-plus/lib/models/git-commit.coffee:80:7)
    at startCommit (/packages/git-plus/lib/models/git-commit.coffee:108:5)
    at /packages/git-plus/lib/models/git-commit.coffee:120:20
```

### Commands

```
     -1:27.6.0 tree-view:reveal-active-file (div.line)
     -1:23.9.0 tree-view:move (li.file.entry.list-item.selected)
     -1:22.9.0 core:select-all (input.hidden-input)
     -1:22.4.0 core:copy (input.hidden-input)
     -1:19.1.0 core:paste (input.hidden-input)
     -1:18.6.0 core:move-right (input.hidden-input)
     -1:18.4.0 core:move-up (input.hidden-input)
     -1:18.1.0 core:select-down (input.hidden-input)
     -1:17.7.0 core:copy (input.hidden-input)
  2x -1:17.5.0 core:paste (input.hidden-input)
     -1:14.4.0 core:delete (input.hidden-input)
  2x -1:10.5.0 core:backspace (input.hidden-input)
     -1:09.2.0 core:save (input.hidden-input)
  6x -0:49.4.0 find-and-replace:select-next (input.hidden-input)
     -0:42.7.0 core:save (input.hidden-input)
     -0:33.2.0 git-plus:add-all-and-commit (input.hidden-input)
```

### Non-Core Packages

```
autocomplete-emojis 2.5.0 
color-picker 2.2.3 
file-icons 2.0.8 
git-plus 7.0.7 
git-tab-status 1.9.2 
git-time-machine 1.5.4 
highlight-line 0.12.0 
highlight-selected 0.11.2 
language-docker 1.1.7 
language-haml 0.24.1 
linter 1.11.21 
linter-docker 0.1.4 
linter-haml 2.0.0 
linter-rubocop 0.5.2 
markdown-scroll-sync 2.1.2 
merge-conflicts 1.4.4 
minimap 4.26.5 
minimap-git-diff 4.3.1 
project-manager 3.3.3 
rails-transporter 1.6.5 
railscast-theme 0.3.0 
```

**Comment by 50Wliu** [comment]
@zzjin @Bjoernsen what were you doing when this error occurred?

**Comment by zzjin** [comment]
@50Wliu just open one dir ,and do project search,seems it only appear at first time cross project search.

**Comment by Bjoernsen** [comment]
@50Wliu it does not happen all the time. 

The first times I had this error, I used Git-Plus to add, commit and push my changes. The error appeared when I typed `STRG + Shift + A + a` and the commit message dialog appeared.

As @zzjin mentioned, the error also (sometimes) appeared when searching the project.

**Comment by adamreisnz** [comment]
Started happening to me a lot as well when trying to find files in project.

**Comment by adamreisnz** [comment]
It might be related to tabs as well. I've had this same error occur when I had a find in project results panel open, and then moved tabs into a separate view and out again. (E.g. split window) It happened on every tab movement while the window was open.

When I looked at the window, no project find results were being displayed. After closing it and searching again, the error appeared again, but results showed up.

After trying again, the error didn't show up anymore and moving tabs was fine.

Tried to find a consistent way to replicate but no luck so far.

**Comment by lgeiger** [comment]
I'm also running into this error randomly on Atom 1.14.0-beta2.

It sometimes occurs during a multi-file search. Unfortunately I don't know how to reliably reproduce this.

I managed to log [`match`](https://github.com/atom/find-and-replace/blob/master/lib/project/match-view.coffee#L13) right before the error occurs:

- A normal multi-file search shows:
<img width="1051" alt="bildschirmfoto 2017-01-20 um 02 45 23" src="https://cloud.githubusercontent.com/assets/13285808/22132471/8a56215e-deba-11e6-90da-f98bf9b9fa12.png">

- And sometimes:
<img width="1057" alt="bildschirmfoto 2017-01-20 um 02 46 11" src="https://cloud.githubusercontent.com/assets/13285808/22132503/ba80cf82-deba-11e6-83a1-6bed55e60e6c.png">
This fails because `lineText` is undefined.

**Comment by as-cii** [comment]
Here's a gif that demonstrates how to consistently reproduce this issue (courtesy of @Ben3eeE ✨):

![splice of undefined 2](https://cloud.githubusercontent.com/assets/482957/22260150/5aeb7b7e-e268-11e6-98e1-86cfab4f6467.gif)

**Comment by as-cii** [comment]
Thanks everyone for the reports and apologies for this regression. A fix for this will land in Atom 1.14.0-beta3, which should be published this week. ⚡️
